pipeline/pipeline.go

```diff
@@ -487,24 +552,31 @@ func (p *Pipeline) maintenance() {
 		p.antispamer.maintenance()
 		p.metricsHolder.maintenance()
 
-		totalCommitted := p.totalCommitted.Load()
-		deltaCommitted := int(totalCommitted - lastCommitted)
-
-		totalSize := p.totalSize.Load()
-		deltaSize := int(totalSize - lastSize)
+		// set new values and get the diffs
+		deltaInputEvents := inputEvents.setAndGetDiff(p.inputEvents.Load())
+		deltaInputSize := inputSize.setAndGetDiff(p.inputSize.Load())
+		deltaOutputEvents := outputEvents.setAndGetDiff(p.outputEvents.Load())
+		deltaOutputSize := outputSize.setAndGetDiff(p.outputSize.Load())
 
-		rate := int(float64(deltaCommitted) * float64(time.Second) / float64(interval))
-		rateMb := float64(deltaSize) * float64(time.Second) / float64(interval) / 1024 / 1024
-
-		tc := totalCommitted
-		if totalCommitted == 0 {
-			tc = 1
+		// increment the metrics
+		getCounter := func(metric string) prometheus.Counter {
+			return stats.GetCounter(p.subsystemName(), metric)
 		}
-
-		p.logger.Infof("%q pipeline stats interval=%ds, active procs=%d/%d, queue=%d/%d, out=%d|%.1fMb, rate=%d/s|%.1fMb/s, total=%d|%.1fMb, avg size=%d, max size=%d", p.Name, interval/time.Second, p.activeProcs.Load(), p.procCount.Load(), p.settings.Capacity-p.eventPool.freeEventsCount, p.settings.Capacity, deltaCommitted, float64(deltaSize)/1024.0/1024.0, rate, rateMb, totalCommitted, float64(totalSize)/1024.0/1024.0, totalSize/tc, p.maxSize)
-
-		lastCommitted = totalCommitted
-		lastSize = totalSize
+		getCounter(inputEventsCountMetric).Add(float64(deltaInputEvents))
```

Comment:
`float64(deltaInputEvents)` can be avoided if returning of `setAndGetDiff` float64

Reply:
done